rest-server/src/util/error.d.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the 'Software'), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import { HttpError } from "http-errors";
+
+declare type Status = 
+    'Bad Request' |
+    'Conflict' |
+    'Forbidden' |
+    'Internal Server Error' |
+    'Not Found' |
+    'Unauthorized';
+declare type Code =
+    'ERR_BAD_CONFIGURATION' |
+    'ERR_CONFLICT_JOB' |
+    'ERR_CONFLICT_USER' |
+    'ERR_FORBIDDEN_USER' |
+    'ERR_INCORRECT_PASSWORD' |
+    'ERR_INVALID_PARAMETERS' |
+    'ERR_NO_API' |
+    'ERR_NO_JOB' |
+    'ERR_NO_JOB_CONFIG' |
+    'ERR_NO_JOB_SSH' |
+    'ERR_NO_USER' |
+    'ERR_NO_VIRTUAL_CLUSTER' |
+    'ERR_REMOVE_ADMIN' |
+    'ERR_UNAUTHORIZED_USER' |
+    'ERR_UNKNOWN';
+
+declare function createError(status: Status, code: Code, message: string): HttpError;
+declare namespace createError {
+    declare function unknown(cause: Error | string): HttpError;
```

Comment:
What does `Error | string` mean please?

Reply:
`Error | string` means union type: either an Error object or a string.

The upstream error may be an error object by utility library, or an HTTP response text.